lib/Dialect/FIRRTL/Transforms/LowerXMR.cpp

```diff
@@ -130,6 +130,20 @@ class LowerXMRPass : public LowerXMRBase<LowerXMRPass> {
             dataFlowClasses.unionSets(connect.getSrc(), connect.getDest());
             return success();
           })
+          .Case<RefSubOp>([&](RefSubOp op) {
+            markForRemoval(op);
+            auto defMem = dyn_cast<MemOp>(op.getInput().getDefiningOp());
+            if (!defMem) {
+              defMem.emitOpError("can only lower RefSubOp of Memory");
+              return failure();
+            }
```

Comment:
Erroring out this late is kind of annoying. Is there a way this check can be moved into a verifier? (Nothing seems obvious to me and this doesn't need to be resolved for this PR.)

Reply:
This error can only be triggered by some issue in the compiler, because the user cannot generate the `RefSubOp`, and they can only remain in the IR after lowering the `MemOp`. We can try to add some default lowering of this in a followup.